vdr/didx509/resolver.go

```diff
@@ -107,14 +108,17 @@ func (r Resolver) Resolve(id did.DID, metadata *resolver.ResolveMetadata) (*did.
 	if err != nil {
 		return nil, nil, fmt.Errorf("did:x509 x5c certificate parsing: %w", err)
 	}
-	_, err = findCertificateByHash(chain, ref.RootCertRef, ref.Method)
+	caFingerprintCert, err := findCertificateByHash(chain, ref.CAFingerprint, ref.Method)
 	if err != nil {
 		return nil, nil, err
 	}
 	validationCert, err := findValidationCertificate(metadata, chain)
 	if err != nil {
 		return nil, nil, err
 	}
+	if bytes.Equal(caFingerprintCert.Raw, validationCert.Raw) {
+		return nil, nil, fmt.Errorf("did:509 ca-fingerprint refers to leaf certificate, must be either root or intermediate CA certificate")
+	}
```

Comment:
Is this requirement true for all did:x509 resolving, or specifically for our use-case?
If so, I think this means the policy file is incorrect: Can we detect that once somewhere else instead of during every resolving action?
(I can't find in our documentation which did:x509 spec we are using so I couldn't look it up)

Reply:
https://trustoverip.github.io/tswg-did-x509-method-specification/#identifier-syntax states:

> ca-fingerprint is chain[i].fingerprint[ca-fingerprint-alg] with i > 0, that is, either an intermediate or root CA certificate.

Although it's kinda cryptic, given x5c is sorted leaf-to-root, I think what they mean (twice) is that it shouldn't be the leaf, but a CA cert.